superset-frontend/src/explore/components/controls/AdhocFilterControl.jsx

```diff
@@ -90,6 +92,46 @@ export default class AdhocFilterControl extends React.Component {
     };
   }
 
+  componentDidMount() {
+    const { datasource } = this.props;
+    if (datasource && datasource.type === 'table') {
+      const dbId = datasource.database ? datasource.database.id : null;
+      const datasourceName = datasource.datasource_name;
+      const datasourceSchema = datasource.schema;
+
+      if (dbId && datasourceName && datasourceSchema) {
+        SupersetClient.get({
+          endpoint: `/superset/extra_table_metadata/${dbId}/${datasourceName}/${datasourceSchema}/`,
+        }).then(
+          ({ json }) => {
+            if (json && json.partitions) {
```

Comment:
it might be worth mentioning that this fails with virtual datasources. i believe we'll get into this if statement but with a 404

Reply:
for virtual table, it doesn't have schema, so won't make the API call to get `extra_table_metadata `

Reply:
when checking a virtual datasource of mine in the bootstrap data, i saw a schema associated with it... maybe a fluke?